### DISCUSSION TITLE
Display on web?

### DISCUSSION THREAD

**Discussion by FlintyLemming** [google/libultrahdr#269] [Q&A] [answered]
Ultra HDR is fantastic. I wonder is there any thing like JavaScript library can let me render fully Ultra HDR photos on web.

**Comment by gregbenz** [comment, marked as answer]
You don't need that, nor can you really do anything about it. Support is either in the browser or is not. 

Chrome, Edge, Brave, and Opera all properly support HDR gain maps now. (note that this does not apply on iOS / iPadOS, which are functionally Safari / WebKit under the hood).

The current versions of Safari and FireFox do not understand gain maps. Because JPG gain maps are 100% backwards compatible, these browsers will just show the base SDR version of your image. So it will look fine, but HDR monitors will not show any benefit on those browsers until they support HDR gain maps.
